dist/pythonlibs/testrunner/spawn.py

```diff
@@ -89,11 +89,18 @@ def setup_child(timeout=10, spawnclass=pexpect.spawnu, env=None, logfile=None):
 
 
 def teardown_child(child):
+    pid = child.pid
     try:
-        os.killpg(os.getpgid(child.pid), signal.SIGKILL)
+        os.killpg(os.getpgid(pid), signal.SIGTERM)
     except ProcessLookupError:
         print("Process already stopped")
-
+    time.sleep(1)
+    # kill still lingering processes
+    try:
+        os.killpg(os.getpgid(pid), signal.SIGKILL)
+    except ProcessLookupError:
+        # This is what we actually wanted
+        pass
```

Comment:
We can avoid the `sleep` in the case the process was already stopped, right?
```suggestion
    else:
       time.sleep(1)
       # kill still lingering processes
       try:
           os.killpg(os.getpgid(pid), signal.SIGKILL)
       except ProcessLookupError:
           # This is what we actually wanted
           pass
```

Reply:
Yeah, right. If the process is already killed when trying to terminate we can be sure its killed and don't need to kill it again ;-)

Reply:
Done